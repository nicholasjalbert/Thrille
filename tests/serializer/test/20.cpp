//test semaphore initialization (deadlock on fail)

#include <pthread.h>
#include <iostream>
#include <stdlib.h>
#include <unistd.h>
#include <stdio.h>
#include <semaphore.h>

using namespace std;

int x;
bool y;
sem_t sem;
pthread_t child, child2;
pthread_mutex_t mutex, mutex2;
pthread_cond_t cond, cond2;

void setup() {
    pthread_mutex_init(&mutex, NULL);
    pthread_mutex_init(&mutex2, NULL);
    pthread_cond_init(&cond, NULL);
    pthread_cond_init(&cond2, NULL);
    sem_init(&sem, 0, 1);
}

void teardown() {
    pthread_mutex_destroy(&mutex);
    pthread_mutex_destroy(&mutex2);
    pthread_cond_destroy(&cond);
    pthread_cond_destroy(&cond2);
}

void * routine(void * arg) {
}


int main(int argc, char *argv[]) {
    setup();
    sem_post(&sem);
    teardown();
}

