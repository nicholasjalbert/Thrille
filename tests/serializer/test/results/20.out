Number of Context Switches: 0
Number of Preemptions: 0
